doc/src/guides/solving/solve-system-of-equations-algebraically.md

```diff
@@ -0,0 +1,130 @@
+-4z)\}}$.
+
+Alternatives to consider:
+- Some systems of equations cannot be solved algebraically (either at all or by
+SymPy), so you may have to {func}`solve your system of equations numerically
+using nsolve() <sympy.solvers.solvers.nsolve>` instead.
+
+Here is an example of solving a system of equations algebraically:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> solve([x**2 + y - 2*z, y + 4*z], x, y, dict=True)
+[{x: -sqrt(6)*sqrt(z), y: -4*z}, {x: sqrt(6)*sqrt(z), y: -4*z}]
+```
+
+## Guidance
+
+Refer to
+[](solving-guidance.md#include-the-variable-to-be-solved-for-in-the-function-call)
+and [](solving-guidance.md#ensure-consistent-formatting-from).
+
+There are two methods below for containing solution results: dictionary or set.
+A dictionary is easier to interrogate programmatically, so if you need to
+extract solutions using code, we recommend the dictionary approach.
+
+## Solve and Use Results in a Dictionary
+
+### Solve Into a Dictionary of Solutions
+
+You can solve a system of equations for some variables (for example, $x$ and
+$y$) leaving another symbol as a variable (for example, $z$). You can specify
+the variables to solve for as multiple separate arguments, or as a list (or
+tuple):
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> equations = [x**2 + y - 2*z, y + 4*z]
+>>> solutions = solve(equations, x, y, dict=True)
+>>> solutions
+[{x: -sqrt(6)*sqrt(z), y: -4*z}, {x: sqrt(6)*sqrt(z), y: -4*z}]
+```
+
+### Use a Dictionary of Solutions
+
+You can then extract solutions by specifying in brackets ("slicing") the
+solution number, and then the symbol. For example `solutions[0][x]` gives the
+result for `x` in the first solution:
+
+```py
+>>> solutions[0][x]
+-sqrt(6)*sqrt(z)
+>>> solutions[0][y]
+-4*z
+```
+
+## Solve Results in a Set
+
+To get a list of symbols and set of solutions, use `set=True` instead of
+`dict=True`:
+
+```py
+from sympy import solve
+from sympy.abc import x, y, z
+solve([x**2 + y - 2*z, y + 4*z], [x, y], set=True)
+([x, y], {(-sqrt(6)*sqrt(z), -4*z), (sqrt(6)*sqrt(z), -4*z)})
+```
+
+## Options That Can Speed up {func}`~.solve`
+
+Refer to [](solving-guidance.md#options-that-can-speed-up).
+
+## Not All Systems of Equations Can be Solved
+
+### Systems of Equations With no Solution
+
+Some systems of equations have no solution. For example, the following system
+reduces to $z = 2z$, which has no general solution:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> solve([x + y - z, x + y - 2*z], [x, y], dict=True)
+[]
+```
+
+The following system is
+[overconstrained](https://en.wikipedia.org/wiki/Overdetermined_system), meaning
+there are more equations (three) than unknowns to be solved for (two, namely $x$
+and $y$), and has no solution:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> solve([x + y - z, x - 2*z, 2*x - y], [x, y], dict=True)
+[]
+```
```

Comment:
Again I think that this example is not suitable for illustrating this point because for some values of `z` (i.e. `z=0`) the system does have solutions.

Reply:
Whether there is a solution for all variables is not the point, though, is it? There are no values for `x` and `y` that will always set these 3 to 0 for arbitrary `z` just like there is no value of `x` that will set both of the following equations to 0 for all `y` (though there is a value of `x` *and* `y` that will do so):
```python
>>> solve([x + y, x - y - 2], x)
[]
>>> solve([x + y, x - y - 2], x, y)
{x: 1, y: -1}
```

Reply:
Oh...there is a difference between the equations and the symbols being requested. 3 equations and 3 unknowns were given in your example, so the system is neither over nor underdetermined. But you requested 2 variables and this treats the 3rd like a constant. And then the idea of "a solution for all values of that constant applies" - there is no such solution so empty list is returned.

Reply:
The terminology that I would use to describe this is that if you do `solve(eqs, [x, y])` then `x` and `y` are the "unknowns". Any other free symbols in the equations are "parameters". A function like `solve` will attempt to return the set of solutions for `x` and `y` under *generic* conditions on the parameters. The solutions returned might be invalid for some *degenerate* cases like `z = 0` but a set of solutions that is only valid under *degenerate* conditions will not be returned. If for almost all possible values of the parameters the system has no solutions then solve will say that it has no solutions.

Reply:
It's also probably good to stress in this and the other solve guides that SymPy doesn't make a distinction between Symbols that are variables or constants (or unknowns or parameters) on the symbol itself, but rather the distinction is made based on how the Symbol is used, e.g., whether or not it is included as one of the symbols to solve for in the second argument to `solve`. It's a common point of confusion where people think they need to somehow define the symbols differently to indicate that they are constant or something, but really they are all just symbols.